### ISSUE TITLE
hyprland/workspaces persistent workspaces don't work

### ISSUE THREAD

**Issue by michal-mnich** [Alexays/Waybar#3470]
After the update to 0.10.4 on Arch Linux, persistent workspaces setting on hyprland is completely ignored. No errors are reported when running waybar from the command line. Sample settings:

```
  "hyprland/workspaces": {
        "format": "{name}",
        "persistent_workspaces": {
            "eDP-1": [ 1, 2, 3 ],
            "HDMI-A-1": [ 4, 5, 6 ],
        }
    },
```

**Comment by Wiian16** [comment]
I'm having the same issue, downgrading to 0.10.3 fixes this. Setting the new flag `special-visible-only` doesn't seem to have any affect as either `true` or `false`.

I'm not very familiar with c++, but looking at commit #3364, `populatePersistentWorkspacesConfig` was removed from `workspaces.hpp` and `workspaces.cpp`. Does anybody know if this could cause this issue?

**Comment by Pierceisgone** [comment]
Hello, this can be fixed by changing `persistent_workspaces` to  `persistent-workspaces` (Change the `_` to a `-`).

**Comment by JasonBoyett** [comment]
I got mine working in a slightly odd way.

### TLDR; I put my options for "hyperland/workspaves" directly after sourcing it

## Non Working version

```jsonc
{
  "position": "top",
  "modules-left": ["hyprland/workspaces"],
   // In the working version I just put the options right here
  "modules-center": ["hyprland/window"],
  "modules-right": ["network", "pulseaudio", "battery", "clock"],
  "clock": {
    "format": "<span foreground='#f5c2e7'>   </span>{:%a %d %H:%M}",
    "tooltip-format": "<big>{:%Y %B}</big>\n<tt><small>{calendar}</small></tt>"
  },
  "battery": {
    "states": {
      "warning": 30,
      "critical": 15
    },
    "hyprland/workspaces": {
      "active-only": false
      "persistent-workspaces": {
        "DP-1": [8, 9, 10],
        "DP-2": [1, 2, 3, 4, 5, 6, 7],
      },
    },
    "format": "<span size='13000' foreground='#a6e3a1'>{icon} </span> {capacity}%",
    "format-warning": "<span size='13000' foreground='#B1E3AD'>{icon} </span> {capacity}%",
    "format-critical": "<span size='13000' foreground='#E38C8F'>{icon} </span> {capacity}%",
    "format-charging": "<span size='13000' foreground='#B1E3AD'> </span>{capacity}%",
    "format-plugged": "<span size='13000' foreground='#B1E3AD'> </span>{capacity}%",
    "format-alt": "<span size='13000' foreground='#B1E3AD'>{icon} </span> {time}",
    "format-full": "<span size='13000' foreground='#B1E3AD'> </span>{capacity}%",
    "format-icons": ["", "", "", "", ""],
    "tooltip-format": "{time}"
  },
  "network": {
    "format-ethernet": "<span size='13000' foreground='#f5e0dc'> </span> Wired",
    "format-wifi": "<span size='13000' foreground='#f5e0dc'>  </span>{essid}",
    "format-linked": "{ifname} (No IP) ",
    "format-disconnected": "<span size='13000' foreground='#f5e0dc'>  </span>Disconnected",
    "tooltip-format-wifi": "Signal Strenght: {signalStrength}%"
  },
  "pulseaudio": {
      "format": "{icon}  {volume}%",
      "format-muted": "",
      "format-icons": {
          "default": ["", "", " "]
      },
      "on-click": "pavucontrol"
  }
}
```

## Working Version
```jsonc
{
  "position": "top",
  "modules-left": ["hyprland/workspaces"],
  "hyprland/workspaces": {
    "active-only": false,
    "persistent-workspaces": {
      "DP-1": [8, 9, 10],
      "DP-2": [1, 2, 3, 4, 5, 6, 7],
    },
  },
  "modules-center": ["hyprland/window"],
  "modules-right": ["network", "pulseaudio", "battery", "clock"],
  "clock": {
    "format": "<span foreground='#f5c2e7'>   </span>{:%a %d %H:%M}",
    "tooltip-format": "<big>{:%Y %B}</big>\n<tt><small>{calendar}</small></tt>"
  },
  "battery": {
    "states": {
      "warning": 30,
      "critical": 15
    },
    // In the non working version the workspace options were here
    "format": "<span size='13000' foreground='#a6e3a1'>{icon} </span> {capacity}%",
    "format-warning": "<span size='13000' foreground='#B1E3AD'>{icon} </span> {capacity}%",
    "format-critical": "<span size='13000' foreground='#E38C8F'>{icon} </span> {capacity}%",
    "format-charging": "<span size='13000' foreground='#B1E3AD'> </span>{capacity}%",
    "format-plugged": "<span size='13000' foreground='#B1E3AD'> </span>{capacity}%",
    "format-alt": "<span size='13000' foreground='#B1E3AD'>{icon} </span> {time}",
    "format-full": "<span size='13000' foreground='#B1E3AD'> </span>{capacity}%",
    "format-icons": ["", "", "", "", ""],
    "tooltip-format": "{time}"
  },
  "network": {
    "format-ethernet": "<span size='13000' foreground='#f5e0dc'> </span> Wired",
    "format-wifi": "<span size='13000' foreground='#f5e0dc'>  </span>{essid}",
    "format-linked": "{ifname} (No IP) ",
    "format-disconnected": "<span size='13000' foreground='#f5e0dc'>  </span>Disconnected",
    "tooltip-format-wifi": "Signal Strenght: {signalStrength}%"
  },
  "pulseaudio": {
      "format": "{icon}  {volume}%",
      "format-muted": "",
      "format-icons": {
          "default": ["", "", " "]
      },
      "on-click": "pavucontrol"
  }
}
```
